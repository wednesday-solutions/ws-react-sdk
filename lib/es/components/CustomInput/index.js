import{_ as r,a as n}from"../_rollupPluginBabelHelpers-c2a46d82.js";import"antd/lib/input/style";import e from"antd/lib/input";import"../_rollup-plugin-inject-process-env-0dfb24f4.js";import t from"react";import{c as o}from"../colors-bf7c26a7.js";import i from"styled-components";import{m as a}from"../media-54d0e07e.js";var p,m=i(e)(p||(p=r(["\n  height: 3.5rem;\n  ","\n  ","\n  &.ant-input, &.ant-input-affix-wrapper {\n    padding-left: 1rem;\n    background-color: ",";\n    border: 1px solid ",";\n    width: 100%;\n    border-radius: 8px;\n    &.ant-form-item-label {\n      font-size: 20px;\n    }\n    &::placeholder {\n      color: ",";\n    }\n  }\n"])),a.desktop.max("\n    height: 3rem;\n  "),a.largeMobile.max("\n    height: 2.5rem;\n  "),o.white,o.gray,o.gray);function l(r){return t.createElement(m,n({"data-testid":"custom-input"},r))}l.propTypes={};export default l;
