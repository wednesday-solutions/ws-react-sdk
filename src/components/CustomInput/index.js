/**
 *
 * CustomInput
 *
 */

import React from 'react';
import { Input } from 'antd';
import colors from '@themes/colors';
import styled from 'styled-components';
import { media } from '@themes';

const CI = styled(Input)`
  height: 3.5rem;
  ${media.desktop.max(`
    height: 3rem;
  `)}
  ${media.largeMobile.max(`
    height: 2.5rem;
  `)}
  &.ant-input, &.ant-input-affix-wrapper {
    padding-left: 1rem;
    background-color: ${colors.white};
    border: 1px solid ${colors.gray};
    width: 100%;
    border-radius: 8px;
    &.ant-form-item-label {
      font-size: 20px;
    }
    &::placeholder {
      color: ${colors.gray};
    }
  }
`;

function CustomInput(props) {
  return <CI data-testid={'custom-input'} {...props} />;
}

CustomInput.propTypes = {};

export default CustomInput;
