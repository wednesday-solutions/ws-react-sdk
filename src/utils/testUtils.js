import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { DEFAULT_LOCALE, translationMessages } from '@src/i18n';
import IntlGlobalProvider from '@components/IntlGlobalProvider';

export const renderWithIntl = children => {
  return render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );
};

export const getComponentStyles = (Component, props = {}) => {
  renderWithIntl(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);
  // eslint-disable-next-line no-underscore-dangle
  return window.getComputedStyle(componentRoots[0])._values;
};

export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export default {
  timeout,
  renderWithIntl,
  getComponentStyles,
  apiResponseGenerator
};
