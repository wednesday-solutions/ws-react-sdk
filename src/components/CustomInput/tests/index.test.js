/**
 *
 * Tests for CustomInput
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import CustomInput from '../index';

describe('<CustomInput />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<CustomInput />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 CustomInput component', () => {
    const { getAllByTestId } = renderWithIntl(<CustomInput />);
    expect(getAllByTestId('custom-input').length).toBe(1);
  });
});
