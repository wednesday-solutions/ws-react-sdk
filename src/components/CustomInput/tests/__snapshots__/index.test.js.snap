// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomInput /> should render and match the snapshot 1`] = `
<body>
  <div>
    <input
      class="ant-input sc-bdfBwQ gZAiAa"
      data-testid="custom-input"
      type="text"
      value=""
    />
  </div>
</body>
`;
